app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/ConfirmYourDecisionController.scala

```diff
@@ -53,34 +45,38 @@ class ConfirmYourDecisionController @Inject()(
   val submissionService: SubmissionService,
-  confirmYourDecisionPage: ConfirmYourDecisionPage,
-  val applicationDeclinedPage: ApplicationDeclinedPage
+  confirmYourDecisionPage: ConfirmYourDecisionPage
 )(implicit override val ec: ExecutionContext) 
     extends AbstractApplicationController(strideAuthConfig, authConnector, forbiddenHandler, mcc, errorHandler) {
       
   import ConfirmYourDecisionController._
 
-  def page(applicationId: ApplicationId): Action[AnyContent] = loggedInWithApplicationAndSubmission(applicationId) { implicit request =>
+  def page(applicationId: ApplicationId) = loggedInWithApplicationAndSubmission(applicationId) { implicit request =>
     if(request.markedSubmission.isFail) 
       successful(Redirect(uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.routes.DeclinedJourneyController.provideReasonsPage(applicationId)))
     else
       successful(Ok(confirmYourDecisionPage(ViewModel(applicationId, request.application.name))))
   }
 
-  def action(applicationId: ApplicationId): Action[AnyContent] = loggedInWithApplicationAndSubmission(applicationId) { implicit request => 
+  def action(applicationId: ApplicationId) = loggedInWithApplicationAndSubmission(applicationId) { implicit request => 
     request.body.asFormUrlEncoded.getOrElse(Map.empty).get("grant-decision").flatMap(_.headOption) match {
       case Some("decline")              => successful(Redirect(uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.routes.DeclinedJourneyController.provideReasonsPage(applicationId)))
+      case Some("grant-with-warnings")  => successful(Redirect(uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.routes.GrantedJourneyController.provideWarningsPage(applicationId)))
       case Some("grant")                => grantAccess(applicationId)
-      // case Some("grant-with-warnings")  => successful(Ok(confirmYourDecisionPage(ViewModel(applicationId, request.application.name))))
-
       case _                            => successful(Redirect(uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.routes.ConfirmYourDecisionController.page(applicationId)))
     }
   }
 
-  private def grantAccess(applicationId: ApplicationId)(implicit request: MarkedSubmissionApplicationRequest[AnyContent]): Future[Result] = {
-    for {
-      _ <- submissionService.grant(applicationId, request.name.get)
-    } yield Redirect(uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.routes.GrantedJourneyController.grantedPage(applicationId).url)
+  private def grantAccess(applicationId: ApplicationId)(implicit request: MarkedSubmissionApplicationRequest[AnyContent]) = {
+    submissionService.grant(applicationId, request.name.get)
+    .map {
+      _ match {
+        case Right(value) => Redirect(uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.routes.GrantedJourneyController.grantedPage(applicationId).url)
+        case Left(err) => {
+          logger.warn(s"Error granting access: $err")
```

Comment:
should this log entry (and the one in the other controller) contain the applicationId?